### PR TITLE
Little refactoring Twig\Helper. The method castMenu inject array opti…

### PR DESCRIPTION
**Pull request by evrinoma**
Hi
The method castMenu inject array options into Menu Builders
Now you could easy use different options to generate Menu with Menu Builders 
{{ knp_menu_render('main', {'group': constant('Evrinoma\\MenuBundle\\Menu\\MenuInterface::DEFAULT_GROUP') }) }}
Test pass on php74

Signed-off-by: Nikolay Nikolaev <evrinoma@gmail.com>

### CODE REVIEW

File: src/Knp/Menu/AbstractFactory.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+use Knp\Menu\Factory\CoreExtension;
+use Knp\Menu\Factory\ExtensionInterface;
+
+/**
+ * Class AbstractFactory
+ *
+ * @package Knp\Menu
+ */
+abstract class AbstractFactory implements FactoryInterface
```
Comment (stof) [inline src/Knp/Menu/AbstractFactory.php:13]: What is this abstract class ? This looks unrelated to the description of this PR

File: src/Knp/Menu/Twig/Helper.php [hunk 2]
```diff
@@ -88,7 +88,7 @@ public function get($menu, array $path = [], array $options = [])
      */
     public function render($menu, array $options = [], $renderer = null)
     {
-        $menu = $this->castMenu($menu);
+        $menu = $this->castMenu($menu, $options);
```
Comment (stof) [inline src/Knp/Menu/Twig/Helper.php:91]: this is now mixing the renderer options and the builder options, which has already been rejected in the past.

Comment (evrinoma) [reply src/Knp/Menu/Twig/Helper.php:91]: that's ok 
but [here](https://symfony.com/doc/current/bundles/KnpMenuBundle/menu_builder_service.html) i have this opportunity.
I would like to build different menu depending on group. How to build it? create a lot different Menu Builders as Services

File: src/Knp/Menu/AbstractFactory.php (hunk 1 above)
Comment (evrinoma) [reply src/Knp/Menu/AbstractFactory.php:13]: I would like to have a serialized menu object for my app or i would like to write custom extension. In first example for this purpose i'll use JMS\Serializer and I should add annotation to MenuItem. I should override MenuItem and create new CustomMenuFactory, but  i can't use logic in new CustomMenuFactory cause this logic is private

that's ok, Should i  create a new PR?

File: src/Knp/Menu/Twig/Helper.php (hunk 2 above)
Comment (evrinoma) [reply src/Knp/Menu/Twig/Helper.php:91]: thanks i've found solution knp_menu_get

File: src/Knp/Menu/AbstractFactory.php (hunk 1 above)
Comment (stof) [reply src/Knp/Menu/AbstractFactory.php:13]: in any case, this should be part of a PR describing your exact needs and explaining what it does, not hidden in a PR talking about refactoring the Twig helper.
and I'd like a clear description of your use case, as I'm not convinced that allowing to replace the item class used by the factory is actually necessary (btw, `JMS\Serializer` supports defining mapping in other formats than annotations, which allows to define mapping for vendor classes)
